spaceapi_server/src/datastore/common.rs

```diff
@@ -13,8 +13,8 @@ pub trait DataStore : Send {
     fn delete(&mut self, key: &str) -> Result<(), DataStoreError>;
 }
 
-/// A datastore wrapped in a Mutex and a Box. Safe for use in multithreaded situations.
-pub type SafeDataStore = Mutex<Box<DataStore>>;
+/// A datastore wrapped in an Arc, a Mutex and a Box. Safe for use in multithreaded situations.
+pub type SafeDataStore = Arc<Mutex<Box<DataStore>>>;
```

Comment:
Because we're now passing around direct references to a SafeDataStore between threads (when creating a handler).